changelog.d/15886.feature

```diff
@@ -0,0 +1 @@
+Allow configuring the set of workers to proxy outbound federation traffic through via `outbound_federation_restricted_to`.
```

Comment:
Thanks for the review @clokep! Since we had to revert the original PR, I've just incorporated this change in the PR to re-introduce everything -> https://github.com/matrix-org/synapse/pull/15913